trace/trace.go

```diff
@@ -0,0 +1,146 @@
+package trace
```

Comment:
A lot of this code seems to be available here almost exactly the same: https://github.com/dapr/dapr/blob/master/pkg/diagnostics/tracing.go

Can you perhaps move that package into dapr/kit and consolidate?